Tests/src/main/java/com/crystal/tests/ex3_IceCreamMachine/IceCreamMachine.java

```diff
@@ -0,0 +1,51 @@
+package com.crystal.tests.ex3_IceCreamMachine;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+public class IceCreamMachine {
+    public String[] ingredients;
+    public String[] toppings;
+
+    public static class IceCream {
+        public String ingredient;
+        public String topping;
+
+        public IceCream(String ingredient, String topping) {
+            this.ingredient = ingredient;
+            this.topping = topping;
+        }
+    }
+
+    public IceCreamMachine(String[] ingredeints, String[] toppings) {
+        this.ingredients = ingredeints;
+        this.toppings = toppings;
+    }
+
+    public List<IceCream> scoops() {
+        return Arrays.stream(ingredients)
+                .flatMap(s -> Arrays.stream(toppings)
+                        .map(s1 -> new IceCream(s, s1)))
+                .collect(Collectors.toList());
+    }
+
+    public static void main(String[] args) {

```

Comment:
Instead of writing main() to check if the application is working as expected, it is expected from you to write JUnit tests as discussed in the testing session.
The comment is applicable to all Exercises

Reply:
Sorry for the delay.
Okey I will try to use JUnit instead of testing on the main. There was some file path exception on JUnit but I have to check that again. (at JUnit it doesn't seem to reorganize the path from the model root, while on the other side of the app it works well. EX(moduleName/src/main/resource/someData) this path works well at other parts of the application, but when it comes to testing it doesn't work, it works only if I get the path from src, but again we will have the same problem with the rest of the app. )
Another important thing to take into consideration is that if I run the tests from the terminal they get executed as aspected.

Reply:
if you can write a test so I can understand the problem you're having, I'll do a checkout and try to fix it for you